static/js/compose_validate.js

```diff
@@ -175,10 +175,15 @@ export function clear_topic_resolved_warning() {
 export function warn_if_topic_resolved() {
     const stream_name = compose_state.stream_name();
     const topic_name = compose_state.topic();
+    const message_content = compose_state.message_content();
 
     const sub = stream_data.get_sub(stream_name);
 
-    if (sub && topic_name.startsWith(message_edit.RESOLVED_TOPIC_PREFIX)) {
+    if (
+        sub &&
+        message_content !== "" &&
+        topic_name.startsWith(message_edit.RESOLVED_TOPIC_PREFIX)
+    ) {
         const error_area = $("#compose_resolved_topic");
 
         if (error_area.html()) {
```

Comment:
I'd also worry that this might cause a flicker, since it seems to close/reopen the warning on every keypress.